components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java

```diff
@@ -177,12 +218,22 @@ public static void setThreadLocalIsManagedThroughSCIMEP(Boolean value) {
     }
 
     public static String getGlobalConsumerId() {
+
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            return IdentityTenantUtil.getTenantDomainFromContext();
```

Comment:
Wondering whether it's good to return tenant domain from privileged carbon context if thread local is not available